### DISCUSSION TITLE
Sleep, Wakeup and fce cycling with the same button

### DISCUSSION THREAD

**Discussion by czKanev** [stm32duino/Arduino_Core_STM32#2364] [Q&A]
Hi everyone,
Im struggling with a simple task. I have a one button LED device. I want to use STM32LowPower lib to put the stm to deepsleep after I cycle through funtioncs and then again to wake up on next push of the button.
But as soon I attach the wakeuppin, the button works anymore properly and try to wakeup the stm every time the button is pushed and do nothing else.

On atmega I used to detach the wakeuppin every time after wakeup and attached it just before sleep. It worked wonderfully.

Is ist possible to disable/detach wakeuppin?
I tried hal_pwr_disablewakeuppin but without any result.
The STM32 I use is F030C8T6TR

Thank You for any advice

**Comment by fpistm** [comment]
Ho @czKanev 
Never tested this use case.
Unfortunately the library does not have detach api.
I could be an enhancement and I guess using the `HAL_PWR_DisableWakeUpPin`.
I would recommend to ensure all wake up IRQ properly clear before entering 
https://github.com/STMicroelectronics/stm32f0xx_hal_driver/blob/115eb1dc87e26ea29f4f2ca58003650381aabba2/Inc/stm32f0xx_hal_pwr.h#L109-L115
Mainly due to the debounce.
